spot:
  joint_states_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 1000

  #Load Joint controllers
  front_left_hip_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_front_left_hip

  front_left_leg_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_front_left_upper_leg

  front_left_foot_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_front_left_lower_leg   

  front_right_hip_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_front_right_hip

  front_right_leg_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_front_right_upper_leg

  front_right_foot_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_front_right_lower_leg

  back_left_hip_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_back_left_hip

  back_left_leg_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_back_left_upper_leg

  back_left_foot_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_back_left_lower_leg   

  back_right_hip_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_back_right_hip

  back_right_leg_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_back_right_upper_leg

  back_right_foot_position_controller:
    type: position_controllers/JointPositionController
    joint: motor_back_right_lower_leg

#PID values for our Joints
#feel free to change them
  # gazebo_ros_control/pid_gains:
  #   motor_front_left_hip        : {p: 100, d: 0.0, i: 5}
  #   motor_front_left_upper_leg  : {p: 300, d: 0.0, i: 8}
  #   motor_front_left_lower_leg  : {p: 300, d: 0.0, i: 8}
  #   motor_front_right_hip       : {p: 100, d: 0.0, i: 5}
  #   motor_front_right_upper_leg : {p: 300, d: 0.0, i: 8}
  #   motor_front_right_lower_leg : {p: 300, d: 0.0, i: 8}
  #   motor_back_left_hip         : {p: 100, d: 0.0, i: 5}
  #   motor_back_left_upper_leg   : {p: 300, d: 0.0, i: 8}
  #   motor_back_left_lower_leg   : {p: 300, d: 0.0, i: 8}
  #   motor_back_right_hip        : {p: 100, d: 0.0, i: 5}
  #   motor_back_right_upper_leg  : {p: 300, d: 0.0, i: 8}
  #   motor_back_right_lower_leg  : {p: 300, d: 0.0, i: 8}

  gazebo_ros_control/pid_gains:
    motor_front_left_hip        : {p: 100, d: 0.01, i: 10}
    motor_front_left_upper_leg  : {p: 100, d: 0.01, i: 10}
    motor_front_left_lower_leg  : {p: 100, d: 0.01, i: 10}
    motor_front_right_hip       : {p: 100, d: 0.01, i: 10}
    motor_front_right_upper_leg : {p: 100, d: 0.01, i: 10}
    motor_front_right_lower_leg : {p: 100, d: 0.01, i: 10}
    motor_back_left_hip         : {p: 100, d: 0.01, i: 10}
    motor_back_left_upper_leg   : {p: 100, d: 0.01, i: 10}
    motor_back_left_lower_leg   : {p: 100, d: 0.01, i: 10}
    motor_back_right_hip        : {p: 100, d: 0.01, i: 10}
    motor_back_right_upper_leg  : {p: 100, d: 0.01, i: 10}
    motor_back_right_lower_leg  : {p: 100, d: 0.01, i: 10}
